signal.go

```diff
@@ -107,6 +109,26 @@ func (s *Signal) WaitForStateMin(state State) <-chan struct{} {
 	return s.waitForWaiter(state, w)
 }
 
+// WaitForStateMinWithCleanup functions identically to WaitForStateMin, but returns a cleanup function in addition
+// so that the caller can cleanup resources if it no longer wants to wait
+func (s *Signal) WaitForStateMinWithCleanup(state State) (<-chan struct{}, func()) {
```

Comment:
The name sounds a bit clumsy to be honest, almost like java :D, but I don't have a better idea and introducing breaking changes just for this seems overkill. The signal implementation never was the finest piece of software in the first place, so maybe let's just do it the way you proposed.

Reply:
Yeah, I struggled a bit with the naming 😆 
If there are more cases where this cleanup is needed down the road, the interface could be simplified and just always provide it.

Reply:
@frairon any idea what's up with the actions failures? It seems to be because it's running from my fork?

Reply:
very sorry for the long delay. Let's just keep it that way and merge it. 
Also, don't worry about the actions, they're quite flaky right now and we'll try to take care of it.